internal/context.go

```diff
@@ -0,0 +1,274 @@
+package internal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+// ContextHandler maps strings to contexts and keeps track of the current context.
+// It has various methods for manipulating and accessing Contexts.
+type ContextHandler struct {
+	Current  string `yaml:"current-context"`
+	Contexts map[string]*Context
+}
+
+// Context represents a context. As of now it only contains information regarding connections.
+// Meaning: engine url, certificates path and any headers.
+// It also keeps a user's comments regarding the context.
+type Context struct {
+	Engine       string
+	Headers      map[string]string
+	Certificates string
+	Comment      string
+}
+
+var contextFilePath = path.Join(userHomeDir(), ".corectl", "contexts.yml")
+
+func CreateContext(contextName, comment string) string {
+	if contextName == "" {
+		FatalError("\"\" is not a valid context name")
+	}
+
+	createContextFileIfNotExist()
+	handler := NewContextHandler()
+
+	if handler.Exists(contextName) {
+		FatalErrorf("context with name '%s' already exists", contextName)
+	}
+
+	context := &Context{}
+	context.Update(&map[string]interface{}{
+		"engine":       viper.GetString("engine"),
+		"headers":      viper.GetStringMapString("headers"),
+		"certificates": viper.GetString("certificates"),
+		"comment":      comment,
+	})
+
+	if err := context.Validate(); err != nil {
+		FatalErrorf("cannot create context '%s': %s", contextName, err.Error())
+	}
+
+	handler.Contexts[contextName] = context
+
+	LogVerbose("Added context with name: " + contextName)
+
+	handler.Current = contextName
+	handler.Save()
+	return contextName
+}
+
+func RemoveContext(contextName string) (string, bool) {
+	handler := NewContextHandler()
+	contextName, wasCurrent := handler.Remove(contextName)
+	handler.Save()
+	return contextName, wasCurrent
+}
+
+func UpdateContext(contextName, comment string) {
+	handler := NewContextHandler()
+	var context *Context
+	if contextName != "" {
+		handler.Exists(contextName)
+		context = handler.Get(contextName)
+	} else {
+		context = handler.GetCurrent()
+	}
+	updated := context.Update(&map[string]interface{}{
+		"engine":       viper.GetString("engine"),
```

Comment:
engine == wsURL?

Reply:
Yes!

Reply:
As engine is the name of the flag and used similarly in config it seems like the best name here as well.